drivers/wifi/nrf700x/osal/hw_if/hal/src/hal_api.c

```diff
@@ -38,29 +38,19 @@ wifi_nrf_hal_rpu_pktram_buf_map_init(struct wifi_nrf_hal_dev_ctx *hal_dev_ctx)
 
 	hal_dev_ctx->addr_rpu_pktram_base_tx = hal_dev_ctx->addr_rpu_pktram_base;
 
-	hal_dev_ctx->addr_rpu_pktram_base_rx = hal_dev_ctx->addr_rpu_pktram_base_tx +
-		(hal_dev_ctx->hpriv->cfg_params.max_tx_frms *
-		 hal_dev_ctx->hpriv->cfg_params.max_tx_frm_sz);
+	hal_dev_ctx->addr_rpu_pktram_base_rx_pool[MAX_NUM_OF_RX_QUEUES - 1] =
+		(hal_dev_ctx->addr_rpu_pktram_base + RPU_PKTRAM_SIZE) -
+		(hal_dev_ctx->hpriv->cfg_params.rx_buf_pool[MAX_NUM_OF_RX_QUEUES - 1].num_bufs *
+		 hal_dev_ctx->hpriv->cfg_params.rx_buf_pool[MAX_NUM_OF_RX_QUEUES - 1].buf_sz);
 
-	hal_dev_ctx->addr_rpu_pktram_base_rx_pool[0] = hal_dev_ctx->addr_rpu_pktram_base_rx;
-
-	for (pool_idx = 1; pool_idx < MAX_NUM_OF_RX_QUEUES; pool_idx++) {
+	for (pool_idx = MAX_NUM_OF_RX_QUEUES - 2; pool_idx >= 0; pool_idx--) {
 		hal_dev_ctx->addr_rpu_pktram_base_rx_pool[pool_idx] =
-			(hal_dev_ctx->addr_rpu_pktram_base_rx_pool[pool_idx - 1] +
-			 (hal_dev_ctx->hpriv->cfg_params.rx_buf_pool[pool_idx - 1].num_bufs *
-			  hal_dev_ctx->hpriv->cfg_params.rx_buf_pool[pool_idx - 1].buf_sz));
+			(hal_dev_ctx->addr_rpu_pktram_base_rx_pool[pool_idx + 1] -
+			 (hal_dev_ctx->hpriv->cfg_params.rx_buf_pool[pool_idx].num_bufs *
+			  hal_dev_ctx->hpriv->cfg_params.rx_buf_pool[pool_idx].buf_sz));
```

Comment:
This is not easy to understand, randomly doing `MAX_NUM_OF_RX_QUEUES - 2` and then `pool_idx +1`, can we please refactor to make it easier to understand.